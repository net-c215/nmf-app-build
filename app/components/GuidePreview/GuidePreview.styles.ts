import { StyleSheet } from "react-native";

export default StyleSheet.create({
  title: {
    paddingTop: 28,
    paddingBottom: 16,
    textAlign: "center"
  },
  listItem: {
    flexDirection: "row",
    flex: 1,
    padding: 16,
    justifyContent: "space-between"
  },
  button: {
    marginTop: 14,
    alignSelf: "center"
  }
});
