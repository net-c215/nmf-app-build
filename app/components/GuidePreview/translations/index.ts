import en from "./en";
import fr from "./fr";
import de from "./de";

interface TranslationKeys {
  GUIDE_PREVIEW_SEE_ALL: string;
}

export { en, fr, de, TranslationKeys };
