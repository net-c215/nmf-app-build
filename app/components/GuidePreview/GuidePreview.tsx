import React from "react";
import { FlatList, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";

import { t } from "../../utils/translations";
import { Button, Text } from "../../components";
import styles from "./GuidePreview.styles";
import { Guide } from "../../types/common-types";
import colors from "../../style/colors";

interface Props {
  title: string;
  listItems: Guide[];
  onPress: (item: Guide) => void;
}

export default function GuidePreview(props: Props) {
  return (
    <React.Fragment>
      <Text.H3 style={styles.title}>{props.title}</Text.H3>
      <FlatList
        data={props.listItems}
        renderItem={({ item }: { item: Guide }) => (
          <TouchableOpacity
            onPress={() => props.onPress(item)}
            style={styles.listItem}
          >
            <Text.Tertiary>{item.title}</Text.Tertiary>
            <Ionicons
              name={"ios-arrow-forward"}
              size={22}
              color={colors.darkLink}
            />
          </TouchableOpacity>
        )}
      />
      <Button.Secondary
        style={styles.button}
        onPress={() => {
          // do nothing.
        }}
        textType={"Secondary"}
      >
        <Text.Secondary bold green center>
          {t("GUIDE_PREVIEW_SEE_ALL")}
        </Text.Secondary>
      </Button.Secondary>
    </React.Fragment>
  );
}
