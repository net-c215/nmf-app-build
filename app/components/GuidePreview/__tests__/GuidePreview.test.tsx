import React from "react";
import renderer from "react-test-renderer";
import GuidePreview from "../";

it("renders correctly TabbedView", () => {
  const tree = renderer
    .create(
      <GuidePreview
        title="Foo"
        listItems={[]}
        onPressItem={() => {
          // do nothing.
        }}
        onPressSeeAll={() => {
          // do nothing.
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
