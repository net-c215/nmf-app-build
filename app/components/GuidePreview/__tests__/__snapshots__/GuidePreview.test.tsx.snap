// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly TabbedView 1`] = `
<RCTScrollView
  ListFooterComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <Text.H3
        style={
          Object {
            "paddingBottom": 16,
            "paddingTop": 28,
            "textAlign": "center",
          }
        }
      >
        Foo
      </Text.H3>
    </View>
    <View
      onLayout={[Function]}
    >
      <Button.Secondary
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "marginTop": 14,
          }
        }
        textType="Secondary"
      >
        <Text.Secondary
          bold={true}
          center={true}
          green={true}
        >
          See All
        </Text.Secondary>
      </Button.Secondary>
    </View>
  </View>
</RCTScrollView>
`;
